src/uct/rocm/base/rocm_base.c

```diff
@@ -501,6 +501,52 @@ ucs_status_t uct_rocm_base_get_link_type(hsa_amd_link_info_type_t *link_type)
     return UCS_OK;
 }
 
+void uct_rocm_base_get_gpu_product(int *gpu_product)
+{
+    char product_name[64];
+    char gfx_name[64];
+    hsa_status_t status;
+
+    /* default value used for all MI200, MI100, and consumer
+       cards */
+    *gpu_product = AMD_GPU_MI200;
+
+    status = hsa_agent_get_info(uct_rocm_base_agents.gpu_agents[0],
+                                (hsa_agent_info_t)
+                                        HSA_AMD_AGENT_INFO_PRODUCT_NAME,
+                                (void*)product_name);
+    if (status != HSA_STATUS_SUCCESS) {
+        ucs_debug("Error in hsa_agent_info %d", status);
+        return;
+    }
+
+    if (NULL != strstr(product_name, "MI300A")) {
+        *gpu_product = AMD_GPU_MI300A;
+        ucs_trace("found MI300A GPU");
```

Comment:
should we print the GPU index or bus id?

Reply:
we are not truly checking for a particular GPU, in fact GPU Id is not passed in into the function. I can hard code that we check GPU zero if you, but there is limited value in that.  We are assuming at the moment that all GPUs on a node are of the same type.

Reply:
ok, maybe document this assumption before line 511, seems hsa_agent_get_info(uct_rocm_base_agents.gpu_agents[0]) is getting info from GPU 0?

Reply:
I added a comment along those lines.